### DISCUSSION TITLE
obsolete

### DISCUSSION THREAD

**Discussion by samantha-xi-chan** [authelia/authelia#7291] [Documentation Feedback]
obsolete

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

To compile the Authelia project, you need to ensure that your environment meets the necessary requirements and that you use the correct scripts provided by the Authelia team.

### Environmental Requirements:
1. **Go**: Version 1.21 or greater (you have Go 1.22.3 which is compatible).
2. **Node.js**: Version 18 or greater (you have v12.22.9, which is below the required version).
3. **pnpm**: This is required for managing Node.js packages.
4. **Docker and Docker Compose**: These are necessary for container management and to run integration tests or suites.

From the error logs you provided, it seems the issue might be related to the Node.js version, which is below the required version. You should upgrade Node.js to at least version 18.

### Steps to Compile:
1. **Upgrade Node.js**: Ensure Node.js is upgraded to version 18 or greater.
2. **Run the bootstrap script**: This script prepares your environment. It seems you've already run this, but ensure all checks pass after upgrading Node.js.
   ```bash
   bash bootstrap.sh
   ```
3. **Build the project**: Use the `authelia-scripts` utility to build the project. After ensuring the bootstrap script runs successfully without errors, you can build Authelia by running:
   ```bash
   authelia-scripts build
   ```

This script will compile both the Go backend and the React frontend of Authelia. Make sure all the dependencies are correctly installed and that the versions match the project's requirements.

If you encounter any further issues, ensure that all environment variables and dependencies are correctly configured as per the project's documentation.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).
